### PR TITLE
feature(nomad): Add spot price for nomad cluster

### PR DESCRIPTION
**Pull request by briantjt**
Copy paste of https://github.com/hashicorp/terraform-aws-nomad/tree/master/modules/nomad-cluster in order to add the `spot_price` argument which was missing

### CODE REVIEW

File: modules/nomad-clients/main.tf [hunk 1]
```diff
@@ -13,7 +13,7 @@ data "aws_vpc" "selected" {
 # --------------------------------------------------------------------------------------------------
 
 module "nomad_clients" {
-  source = "github.com/hashicorp/terraform-aws-nomad//modules/nomad-cluster?ref=v0.5.0"
```
Comment (qbiqing) [inline modules/nomad-clients/main.tf:16]: I wonder if we should continue to keep track of the repo to take advantage of future releases 🤔  @guangie88 what do you think?

Comment (guangie88) [reply modules/nomad-clients/main.tf:16]: Can just put the above URL back as a comment to remind ourselves I guess.

**Comment by guangie88** [comment]
Can you add a README to `modules/nomad-clients`, don't have to copy and paste from the original README in https://github.com/hashicorp/terraform-aws-nomad/tree/master/modules/nomad-cluster, but just state that it's a copy and paste to support spot instances?

Do include in the fairly standard INOUT section at the end of the README too:

```md
## Inputs and Outputs

Refer to [INOUT.md](INOUT.md)

```

**Review by guangie88 — APPROVED** [review]
LGTM

**Pull request merged**